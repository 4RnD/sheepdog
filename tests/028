#!/bin/bash

# Test sheep recovery logic

seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1        # failure is the default!

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_cleanup

for i in `seq 0 1`; do _start_sheep $i; done

_wait_for_sheep 2

$COLLIE cluster format -c 2

$COLLIE vdi create test 20M

dd if=/dev/urandom | $COLLIE vdi write test
$COLLIE vdi read test | md5sum > /tmp/csum.0

for i in 2 3; do
    _start_sheep $i
    sleep 1
    _kill_sheep $(($i-2))
    _wait_for_sheep_recovery $i
done

dd if=/dev/urandom | $COLLIE vdi write test -p 7002
$COLLIE vdi read test -p 7002 | md5sum > /tmp/csum.1

for i in 2 3; do
    _start_sheep $(($i-2))
    sleep 1
    _kill_sheep $i
    _wait_for_sheep_recovery $(($i-2))
done

$COLLIE vdi read test | md5sum > /tmp/csum.2
$COLLIE vdi read test -p 7001 | md5sum > /tmp/csum.3

$COLLIE vdi object test

for i in `seq 0 4`; do
    $COLLIE vdi object -i $i test
done

if [ "$(diff -u /tmp/csum.0 /tmp/csum.1)" ]; then
    echo "update test successfully"
fi

diff -u /tmp/csum.1 /tmp/csum.2
diff -u /tmp/csum.1 /tmp/csum.3
