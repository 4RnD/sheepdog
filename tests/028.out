QA output created by 028
using backend farm store
Looking for the inode object 0x7c2b25 with 2 nodes

127.0.0.1:7000 has the object (should be 2 copies)
127.0.0.1:7001 has the object (should be 2 copies)
Looking for the object 0x7c2b2500000000 (the inode vid 0x7c2b25 idx 0) with 2 nodes

127.0.0.1:7000 has the object (should be 2 copies)
127.0.0.1:7001 has the object (should be 2 copies)
Looking for the object 0x7c2b2500000001 (the inode vid 0x7c2b25 idx 1) with 2 nodes

127.0.0.1:7000 has the object (should be 2 copies)
127.0.0.1:7001 has the object (should be 2 copies)
Looking for the object 0x7c2b2500000002 (the inode vid 0x7c2b25 idx 2) with 2 nodes

127.0.0.1:7000 has the object (should be 2 copies)
127.0.0.1:7001 has the object (should be 2 copies)
Looking for the object 0x7c2b2500000003 (the inode vid 0x7c2b25 idx 3) with 2 nodes

127.0.0.1:7000 has the object (should be 2 copies)
127.0.0.1:7001 has the object (should be 2 copies)
Looking for the object 0x7c2b2500000004 (the inode vid 0x7c2b25 idx 4) with 2 nodes

127.0.0.1:7000 has the object (should be 2 copies)
127.0.0.1:7001 has the object (should be 2 copies)
update test successfully
